backend/src/main/kotlin/com/example/backend/service/RouteService.kt

```
@@ -67,4 +71,26 @@ class RouteService(
 
         return result
     }
+
+    fun saveRouteLandmark(landmarkRequest: LandmarkDto, routeId: Long): Landmark {
+        return landmarkRepository.save(
+            Landmark(
+                description = landmarkRequest.description,
+                address = landmarkRequest.address,
+                latitude = landmarkRequest.latitude,
+                longitude = landmarkRequest.longitude,
+                route = getRouteById(routeId)
+            )
+        )
+    }
+
+    fun getRouteLandmarksByRouteId(routeId: Long): List<Landmark> {
+        val result = landmarkRepository.findAll().filter { it.route.id == routeId }
+
+        if (result.isEmpty()) {
+            throw EntityNotFoundException()
```

Do we want to throw an exception if there are no landmarks for a route? Are landmarks going to be required, or an optional field? If they are not required it would be fine if we just return an empty list, like for getAllRoutes().

The filter functionality works, but there also might be findAllLandmarksByRouteId already available through the landmarkRepository, it auto implements a ton of different query combos with Hibernate.
In this case it's not a big deal, but just something for everyone to keep in mind in case there is more complex retrievals in the future.